app/app.js

```diff
@@ -0,0 +1,56 @@
+import {App, IonicApp, Platform} from 'ionic/ionic';
+
+import {HelloIonicPage} from './pages/hello-ionic/hello-ionic';
+import {ListPage} from './pages/list/list';
+
+@App({
+  templateUrl: 'build/app.html'
+})
+class MyApp {
+  constructor(app: IonicApp, platform: Platform) {
+
+    // set up our app
+    this.app = app;
+    this.platform = platform;
+    this.initializeApp();
+
+    // set our app's pages
+    this.pages = [
+      { title: 'Hello Ionic', component: HelloIonicPage },
+      { title: 'My First List', component: ListPage }
+    ];
+
+    // make HelloIonicPage the root (or first) page
+    this.rootPage = HelloIonicPage;
+  }
+
+  initializeApp() {
+    this.platform.ready().then(() => {
+      console.log('Platform ready');
+
+      // The platform is now ready. Note: if this callback fails to fire, follow
+      // the Troubleshooting guide for a number of possible solutions:
+      //
+      // Okay, so the platform is ready and our plugins are available.
+      // Here you can do any higher level native things you might need.
+      //
+      // First, let's hide the keyboard accessory bar (only works natively) since
+      // that's a better default:
+      //
+      //
+      // For example, we might change the StatusBar color. This one below is
+      // good for light backgrounds and dark text;
+      if (window.StatusBar) {
+        window.StatusBar.styleDefault();
+      }
+    });
+  }
```

Comment:
This function then calls the `platform.ready()` function which will return a promise when the app is ready. The `initializeApp` function basically just indicates that the app is fully loaded and ready to do stuff, so it’s a safe space to start doing things like calling PhoneGap plugins like the StatusBar plugin. Since this plugin is only available when running through a device, it’ll throw an error when you try running it through the browser.